File: 2025-09-25 - Deques/Exemplo_Python/Node.java
// ------------------------------------------------------------------------
// Node.java
// Classe que representa um nó da lista de prioridade
// Implementa Comparable para ser usada em PriorityQueue
// ------------------------------------------------------------------------
public class Node implements Comparable<Node> {

   private String codigoDisciplina;
   private String nomeDisciplina;
   private int cargaHoraria;
   private int semestre;

   // ---------------------------------------------------------------------
   // Construtor
   // ---------------------------------------------------------------------
   public Node(String codigoDisciplina, String nomeDisciplina, int cargaHoraria, int semestre) {
      this.codigoDisciplina = codigoDisciplina;
      this.nomeDisciplina = nomeDisciplina;
      this.cargaHoraria = cargaHoraria;
      this.semestre = semestre;
   }

   // ---------------------------------------------------------------------
   // Getters e Setters
   // ---------------------------------------------------------------------
   public String getCodigoDisciplina() { return codigoDisciplina; }
   public void setCodigoDisciplina(String codigoDisciplina) { this.codigoDisciplina = codigoDisciplina; }

   public String getNomeDisciplina() { return nomeDisciplina; }
   public void setNomeDisciplina(String nomeDisciplina) { this.nomeDisciplina = nomeDisciplina; }

   public int getCargaHoraria() { return cargaHoraria; }
   public void setCargaHoraria(int cargaHoraria) { this.cargaHoraria = cargaHoraria; }

   public int getSemestre() { return semestre; }
   public void setSemestre(int semestre) { this.semestre = semestre; }

   // ---------------------------------------------------------------------
   // Implementação do Comparable para prioridade (menor semestre = maior prioridade)
   // ---------------------------------------------------------------------
   @Override
   public int compareTo(Node outro) {
      return Integer.compare(this.semestre, outro.semestre);
   }

   @Override
   public String toString() {
      return codigoDisciplina + " – " + nomeDisciplina + " (" + cargaHoraria + "h) - Semestre " + semestre;
   }
}

File: 2025-09-25 - Deques/Exemplo_Python/Node.py
# ------------------------------------------------------------------------
# node.py
# Classe que representa o nó da Deque (lista duplamente ligada)
# ------------------------------------------------------------------------
class Node:
   # ---------------------------------------------------------------------
   # Construtor
   # ---------------------------------------------------------------------
   def __init__(self, codigoDisciplina: str, nomeDisciplina: str, cargaHoraria: int):
      self.__codigoDisciplina = codigoDisciplina
      self.__nomeDisciplina = nomeDisciplina
      self.__cargaHoraria = cargaHoraria
      self.__anterior = None
      self.__proximo = None

   # ---------------------------------------------------------------------
   # Getters e Setters (usando property em Python)
   # ---------------------------------------------------------------------
   @property
   def codigoDisciplina(self):
      return self.__codigoDisciplina

   @codigoDisciplina.setter
   def codigoDisciplina(self, valor):
      self.__codigoDisciplina = valor

   @property
   def nomeDisciplina(self):
      return self.__nomeDisciplina

   @nomeDisciplina.setter
   def nomeDisciplina(self, valor):
      self.__nomeDisciplina = valor

   @property
   def cargaHoraria(self):
      return self.__cargaHoraria

   @cargaHoraria.setter
   def cargaHoraria(self, valor):
      self.__cargaHoraria = valor

   @property
   def anterior(self):
      return self.__anterior

   @anterior.setter
   def anterior(self, valor):
      self.__anterior = valor

   @property
   def proximo(self):
      return self.__proximo

   @proximo.setter
   def proximo(self, valor):
      self.__proximo = valor

   # ---------------------------------------------------------------------
   # Representação em string
   # ---------------------------------------------------------------------
   def __str__(self):
      return f"{self.__codigoDisciplina} – {self.__nomeDisciplina} ({self.__cargaHoraria}h)"
